w2c2/w2c2_base.h

```diff
@@ -52,6 +52,12 @@ typedef double F64;
 #define W2C2_LL(x) x ## ll
 #endif
 
+#if defined(__clang__) && __clang_major__ <= 12
+#define W2C2_PRE_BB do{__asm__ volatile()}while(0)
+#else
+#define W2C2_PRE_BB do{}while(0)
+#endif
+
```

Comment:
- Maybe rename `W2C2_PRE_BB` to something more specific, e.g. `W2C2_LOOP_START`
- Is the `do{ ...} while(0)` wrapper necessary here? It's just a single statement
- Can the define in the else case just be nothing? Loop expressions are fairly common, and targets other than the affected ones should not have to pay the price in code size increase
- Maybe add a short comment explaining why this is needed

Maybe this can be e.g. 
```suggestion
// Prevent LLVM/Clang <=12 from optimizing away infinite loops
#if defined(__clang__) && __clang_major__ <= 12
#define W2C2_LOOP_START __asm__ volatile();
#else
#define W2C2_LOOP_START
#endif
```

Reply:
Thanks for suggesting; applied